### PR TITLE
feat(object): give Individuals a vision as vector

### PR DESCRIPTION
**Pull request by anarchuser**
depends on #28 and #30 

It also uses a specific `Angle` struct which wraps angles around so that periods match.
Using `Angle`, all values within [0; 360) / [0;2 * PI)

### CODE REVIEW

File: src/model/algebra/point/point.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+Point Point::operator - (const Point & p) const {return Point(x - p.x, y - p.y);}
+Point Point::operator - (const Vector & v) const {return * this + -v;}
+
+Vector Point::operator >= (const Point & p) const {
+    const Point q(* this - p);
+    return Vector(atan(q.y / q.x), ORIGIN / q);
+}
+Vector Point::operator <= (const Point & p) const {return p >= * this;}
+
+bool Point::withinCone (const Point & origin, const Vector & center, double angle) const {
```
Comment (xeruf) [inline src/model/algebra/point/point.cpp]: needs docs

Comment (xeruf) [reply src/model/algebra/point/point.cpp]: for each param

File: test/model/algebra/vector/vector.cpp [hunk 2]
```diff
@@ -3,5 +3,6 @@
 #include <catch2/catch.hpp>
 #include <vector>
 
+// TODO
```
Comment (xeruf) [inline test/model/algebra/vector/vector.cpp]: I would've put the TODO inside the empty test, but whatever ^^

Comment (anarchuser) [reply test/model/algebra/vector/vector.cpp]: removed

File: src/model/algebra/point/point.cpp (hunk 1 above)
Comment (anarchuser) [reply src/model/algebra/point/point.cpp]: comes in a future update (upcoming PR)

File: src/model/algebra/algebra.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#ifndef EVE_ANGLE_VECTOR_H
```
Comment (xeruf) [inline src/model/algebra/algebra.h]: this header is a little messed up

File: test/model/algebra/vector/vector.cpp (hunk 2 above)
Comment (xeruf) [reply test/model/algebra/vector/vector.cpp]: no?

**Review by xeruf — CHANGES_REQUESTED** [review]

File: test/model/algebra/vector/vector.cpp (hunk 2 above)
Comment (anarchuser) [reply test/model/algebra/vector/vector.cpp]: yes

File: src/model/algebra/algebra.h (hunk 3 above)
Comment (anarchuser) [reply src/model/algebra/algebra.h]: fixed

**Review by xeruf — APPROVED** [review]

**Pull request merged**